rust/osm2lanes/src/transform.rs

```diff
@@ -622,51 +616,67 @@ fn assemble_ltr(
     }
 }
 
-// See https://wiki.openstreetmap.org/wiki/Proposed_features/cycleway:separation#Typical_values.
-// Lots of these mappings are pretty wacky right now. We need more BufferTypes.
-fn osm_separation_type(x: &str) -> Option<BufferType> {
-    match x {
-        "bollard" | "vertical_panel" => Some(BufferType::FlexPosts),
-        "kerb" | "separation_kerb" => Some(BufferType::Curb),
-        "grass_verge" | "planter" | "tree_row" => Some(BufferType::Planters),
-        "guard_rail" | "jersey_barrier" | "railing" => Some(BufferType::JerseyBarrier),
-        // TODO Make sure there's a parking lane on that side... also mapped? Any flex posts in
-        // between?
-        "parking_lane" => None,
-        "barred_area" | "dashed_line" | "solid_line" => Some(BufferType::Stripes),
-        _ => None,
-    }
-}
-
-pub fn lanes_to_tags(lanes: &[LaneSpec], cfg: &Config) -> Result<Tags, LaneSpecError> {
+pub fn lanes_to_tags(lanes: &[Lane], cfg: &Config) -> Result<Tags, LaneSpecError> {
     let mut tags = Tags::default();
-    let mut oneway = false;
+    let mut _oneway = false;
     tags.insert("highway", "yes"); // TODO, what?
     {
         let lane_count = lanes
             .iter()
             .filter(|lane| {
-                lane.lane_type == LaneType::Driving
-                    || lane.lane_type == LaneType::SharedLeftTurn
-                    || lane.lane_type == LaneType::Bus
+                matches!(
+                    lane,
+                    Lane::Travel {
+                        designated: LaneDesignated::Motor | LaneDesignated::Bus,
+                        ..
+                    }
+                )
             })
             .count();
         tags.insert("lanes", lane_count.to_string());
     }
     // Oneway
     if lanes
         .iter()
-        .filter(|lane| lane.lane_type == LaneType::Driving)
-        .all(|lane| lane.direction == Direction::Forward)
+        .filter(|lane| {
+            matches!(
+                lane,
+                Lane::Travel {
+                    designated: LaneDesignated::Motor,
+                    ..
+                }
+            )
+        })
+        .all(|lane| {
+            matches!(
+                lane,
+                Lane::Travel {
+                    direction: Some(LaneDirection::Forward),
+                    ..
+                }
+            )
+        })
     {
         tags.insert("oneway", "yes");
-        oneway = true;
+        _oneway = true;
     }
     // Pedestrian
     {
         match (
-            lanes.first().unwrap().lane_type == LaneType::Sidewalk,
-            lanes.last().unwrap().lane_type == LaneType::Sidewalk,
+            matches!(
+                lanes.first().unwrap(),
+                Lane::Travel {
```

Comment:
Similar to the `Lane::foot()` constructor, I wonder if we need `lane.is_foot()` predicates to shorten all this boilerplate? Just an idea going forward, not sure. By making the schema more flexible, it'll mean every language implementation will grow its own helper methods like this, which might diverge over time